### ISSUE TITLE
LuigiClient goBack() not working within a Micro App with internal routes

### ISSUE THREAD

**Issue by pixx3l** [SAP/luigi#265]
**Description**

Unable to use the `luigiClient.linkManager().goBack()` feature.

**Implementation**
`luigi-client: 0.3.8`

```typescript
export class LuigiContextService {
  private luigiClient: LuigiClient = LuigiClient;

  goBack() {
    this.luigiClient
      .linkManager()
      .fromContext('customer-recovery-reporting')
      .goBack();
  }

  hasBack() {
    return this.luigiClient
      .linkManager()
      .fromContext('customer-recovery-reporting')
      .hasBack();
  }
}
```

Navigating to a sub-page

```typescript
@Component({
  selector: 'app-table-cell',
  template: `
    <a (click)="goToCampaign(link)">{{value}}</a>
  `,
  encapsulation: ViewEncapsulation.None // allow FD style overwrites
})
export class TableCellComponent {
  constructor(private luigiContext: LuigiContextService) {}

  @Input()
  link: string = null;

  goToCampaign(link) {
    this.luigiContext.goTo(link); // <--- should push route to history
  }
}
```

Usage in app file

```typescript
export class ReportComponent {

  constructor(private luigiContext: LuigiContextService) {}

  // always false
  checkBack() {
    console.log('has back: ', this.luigiContext.hasBack());
  }

  // does nothing
  handleBackButtonClick(): void {
    this.luigiContext.goBack(); <--- should go back to the page TableCellComponent was clicked from
  }
}
```

**Expected result**

Navigating to a page that contains a back button, I expect the last route prior to the current one to load when the button is clicked.

**Actual result**

Nothing. The 'hasBack' check always comes up false when navigating to a page with the back button, and executing the back function does not route

**Steps to reproduce**

N/A - Let me know if you need me to publish the stash branch for reproduction

**Troubleshooting**

* Tried to load the LuigiClient independently into each component using it.
* Trued abstracting the LuigiClient into a single root service, in an effort to make it a singleton in case that has something to do with the navigation tracking.
* Tried running the go back function with and without context.

**Possible Related Issues**
https://github.com/kyma-project/luigi/issues/195

**Comment by pixx3l** [comment]
Update: I can get the `goBack()` function to work if I do my link routing like so: `navigate(path, null, true) But I noticed that when I use this invocation that multiple iframes get created for rendering pages and that seems like overkill to me. *I still have not been able to get `goBack()` to work in a single iframe*

**Comment by maxmarkus** [comment]
Does currently not work if default child node feature is triggered, will be fixed with #216.
